### PR TITLE
add more CDATA-related tests for `Magento\Framework\Config\Dom::merge` and fix failing ones

### PR DESCRIPTION
**Pull request by enl**
### Description

This PR fixes several issues I found trying to update field comment of another module in Magento 2 Admin:

```xml
<comment>
    <![CDATA[
        "Redirect" redirects the client directly to the target store. <br />
        "Countries Popup" shows the Countries Popup to select target store. <br />
        "Auto" does detect when it's needed to use "Redirect" or "Countries Popup" depending on current request. <br />
    ]]>
</comment>
```
If I create another `system.xml` file updating such a comment it will never be replaced because `$this->_isTextNode($matchingNode)` will always return `false`: this `<comment>` node contains three childNodes, none of them is `\DomElement` (empty text node, cdata section, empty text node).

During investigation of this, I created a bunch of new test cases for `Magento\Framework\Config\Dom::merge` and fixed failing ones.

### Manual testing scenarios (*)
Detailed steps described in [THIS comment](https://github.com/magento/magento2/pull/18336#issuecomment-471598656)

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18336) <br/>All committers have signed the CLA.

**Comment by enl** [comment]
The build is failing with quite strange message:

```
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
apt-get.diagnostics
apt-get install failed
$ cat ${TRAVIS_HOME}/apt-get-update.log
Ign:1 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty InRelease
Get:2 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates InRelease [65.9 kB]
Get:3 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports InRelease [65.9 kB]
Get:4 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty Release [58.5 kB]
Get:5 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty Release.gpg [933 B]
Ign:6 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 InRelease
Get:7 http://security.ubuntu.com/ubuntu trusty-security InRelease [65.9 kB]
Get:9 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 Release [2,495 B]
Get:10 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main Sources [521 kB]
Get:11 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 Release.gpg [801 B]
Get:12 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/restricted Sources [6,449 B]
Ign:8 http://toolbelt.heroku.com/ubuntu ./ InRelease
Get:13 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/universe Sources [266 kB]
Get:14 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/multiverse Sources [7,285 B]
Get:15 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty InRelease [15.4 kB]
Get:16 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 Packages [1,380 kB]
Get:17 http://toolbelt.heroku.com/ubuntu ./ Release [1,609 B]
Get:19 http://apt.postgresql.org/pub/repos/apt trusty-pgdg InRelease [61.4 kB]
Ign:20 http://dl.google.com/linux/chrome/deb stable InRelease
Get:22 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main i386 Packages [1,301 kB]
Get:21 http://toolbelt.heroku.com/ubuntu ./ Release.gpg [473 B]
Get:23 http://dl.google.com/linux/chrome/deb stable Release [1,189 B]
Get:24 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main Translation-en [665 kB]
Get:25 https://dl.hhvm.com/ubuntu trusty InRelease [2,411 B]
Get:26 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4/multiverse amd64 Packages [12.4 kB]
Get:27 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/restricted amd64 Packages [21.4 kB]
Get:28 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/restricted i386 Packages [21.1 kB]
Get:29 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/restricted Translation-en [3,704 B]
Get:30 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/universe amd64 Packages [629 kB]
Get:31 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/universe i386 Packages [613 kB]
Get:32 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/universe Translation-en [322 kB]
Ign:33 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty InRelease
Get:34 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/multiverse amd64 Packages [16.0 kB]
Get:35 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/multiverse i386 Packages [16.5 kB]
Get:36 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/multiverse Translation-en [7,680 B]
Get:37 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/main Sources [10.4 kB]
Get:38 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/restricted Sources [40 B]
Get:39 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/universe Sources [41.3 kB]
Get:40 https://download.docker.com/linux/ubuntu trusty InRelease [26.5 kB]
Get:41 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/multiverse Sources [1,747 B]
Get:42 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/main amd64 Packages [14.7 kB]
Get:43 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/main i386 Packages [14.7 kB]
Get:44 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/main Translation-en [7,426 B]
Get:45 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/restricted amd64 Packages [40 B]
Get:46 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/restricted i386 Packages [40 B]
Get:47 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/restricted Translation-en [40 B]
Get:48 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/universe amd64 Packages [52.5 kB]
Get:49 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/universe i386 Packages [52.4 kB]
Get:50 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/universe Translation-en [40.0 kB]
Get:51 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/multiverse amd64 Packages [1,392 B]
Get:52 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/multiverse i386 Packages [1,376 B]
Get:53 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports/multiverse Translation-en [1,165 B]
Get:54 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main Sources [1,335 kB]
Get:55 http://dl.google.com/linux/chrome/deb stable Release.gpg [819 B]
Get:56 http://security.ubuntu.com/ubuntu trusty-security/main Sources [208 kB]
Get:57 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/restricted Sources [5,335 B]
Get:58 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/universe Sources [7,926 kB]
Get:59 http://security.ubuntu.com/ubuntu trusty-security/restricted Sources [5,050 B]
Get:60 http://ppa.launchpad.net/git-core/ppa/ubuntu trusty InRelease [15.4 kB]
Get:61 http://security.ubuntu.com/ubuntu trusty-security/universe Sources [102 kB]
Get:62 http://security.ubuntu.com/ubuntu trusty-security/multiverse Sources [3,072 B]
Get:63 http://security.ubuntu.com/ubuntu trusty-security/main amd64 Packages [958 kB]
Get:64 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/multiverse Sources [211 kB]
Get:65 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main amd64 Packages [1,743 kB]
Get:66 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main i386 Packages [1,739 kB]
Get:67 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/restricted amd64 Packages [16.0 kB]
Get:68 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/restricted i386 Packages [16.4 kB]
Get:69 http://security.ubuntu.com/ubuntu trusty-security/main i386 Packages [883 kB]
Get:70 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/universe amd64 Packages [7,589 kB]
Get:71 http://security.ubuntu.com/ubuntu trusty-security/main Translation-en [505 kB]
Get:72 http://ppa.launchpad.net/openjdk-r/ppa/ubuntu trusty InRelease [15.4 kB]
Get:73 http://security.ubuntu.com/ubuntu trusty-security/restricted amd64 Packages [18.1 kB]
Get:74 http://security.ubuntu.com/ubuntu trusty-security/restricted i386 Packages [17.8 kB]
Get:75 http://security.ubuntu.com/ubuntu trusty-security/restricted Translation-en [3,272 B]
Get:76 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/universe i386 Packages [7,597 kB]
Get:77 http://security.ubuntu.com/ubuntu trusty-security/universe amd64 Packages [333 kB]
Get:79 https://packagecloud.io/computology/apt-backport/ubuntu trusty InRelease [23.7 kB]
Get:80 http://security.ubuntu.com/ubuntu trusty-security/universe i386 Packages [319 kB]
Get:78 http://toolbelt.heroku.com/ubuntu ./ Packages [636 B]
Get:81 http://security.ubuntu.com/ubuntu trusty-security/universe Translation-en [178 kB]
Get:82 http://security.ubuntu.com/ubuntu trusty-security/multiverse amd64 Packages [4,726 B]
Get:18 http://dl.bintray.com/apache/cassandra 39x InRelease [3,168 B]
Get:83 http://security.ubuntu.com/ubuntu trusty-security/multiverse i386 Packages [4,880 B]
Get:84 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/multiverse amd64 Packages [169 kB]
Get:85 https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease [23.2 kB]
Get:86 http://security.ubuntu.com/ubuntu trusty-security/multiverse Translation-en [2,426 B]
Get:87 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/multiverse i386 Packages [172 kB]
Get:88 http://ppa.launchpad.net/pollinate/ppa/ubuntu trusty InRelease [15.4 kB]
Get:89 https://dl.hhvm.com/ubuntu trusty/main amd64 Packages [1,815 B]
Get:90 https://download.docker.com/linux/ubuntu trusty/stable amd64 Packages [4,058 B]
Get:91 https://download.docker.com/linux/ubuntu trusty/edge amd64 Packages [5,282 B]
Get:92 https://packagecloud.io/rabbitmq/rabbitmq-server/ubuntu trusty InRelease [23.7 kB]
Get:93 http://dl.google.com/linux/chrome/deb stable/main amd64 Packages [1,122 B]
Get:94 http://apt.postgresql.org/pub/repos/apt trusty-pgdg/main amd64 Packages [190 kB]
Get:95 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty InRelease [15.5 kB]
Get:96 https://packagecloud.io/computology/apt-backport/ubuntu trusty/main Sources [843 B]
Get:98 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty Release [15.1 kB]
Get:99 https://packagecloud.io/computology/apt-backport/ubuntu trusty/main amd64 Packages [3,625 B]
Get:101 https://packagecloud.io/computology/apt-backport/ubuntu trusty/main i386 Packages [848 B]
Get:102 http://apt.postgresql.org/pub/repos/apt trusty-pgdg/main i386 Packages [189 kB]
Get:103 https://packagecloud.io/github/git-lfs/ubuntu trusty/main Sources [20 B]
Get:104 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main amd64 Packages [1,710 B]
Get:105 https://packagecloud.io/github/git-lfs/ubuntu trusty/main amd64 Packages [7,055 B]
Get:106 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main i386 Packages [1,702 B]
Get:107 https://packagecloud.io/github/git-lfs/ubuntu trusty/main i386 Packages [6,816 B]
Get:108 https://packagecloud.io/rabbitmq/rabbitmq-server/ubuntu trusty/main Sources [20 B]
Get:109 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main Translation-en [933 B]
Get:97 http://dl.bintray.com/apache/cassandra 39x/main amd64 Packages [682 B]
Get:110 https://packagecloud.io/rabbitmq/rabbitmq-server/ubuntu trusty/main amd64 Packages [5,911 B]
Get:111 http://ppa.launchpad.net/git-core/ppa/ubuntu trusty/main amd64 Packages [3,483 B]
Get:112 https://packagecloud.io/rabbitmq/rabbitmq-server/ubuntu trusty/main i386 Packages [5,911 B]
Get:113 http://ppa.launchpad.net/git-core/ppa/ubuntu trusty/main i386 Packages [3,484 B]
Get:114 http://ppa.launchpad.net/git-core/ppa/ubuntu trusty/main Translation-en [2,332 B]
Get:115 http://ppa.launchpad.net/openjdk-r/ppa/ubuntu trusty/main amd64 Packages [7,494 B]
Get:100 http://dl.bintray.com/apache/cassandra 39x/main i386 Packages [682 B]
Get:116 http://ppa.launchpad.net/openjdk-r/ppa/ubuntu trusty/main i386 Packages [7,640 B]
Get:117 http://ppa.launchpad.net/openjdk-r/ppa/ubuntu trusty/main Translation-en [2,388 B]
Get:118 http://ppa.launchpad.net/pollinate/ppa/ubuntu trusty/main amd64 Packages [430 B]
Get:119 http://ppa.launchpad.net/pollinate/ppa/ubuntu trusty/main i386 Packages [430 B]
Get:120 http://ppa.launchpad.net/pollinate/ppa/ubuntu trusty/main Translation-en [374 B]
Get:121 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty/main amd64 Packages [1,551 B]
Get:122 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty/main i386 Packages [1,551 B]
Get:123 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty/main Translation-en [834 B]
Get:124 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty Release.gpg [316 B]
Get:125 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty/main amd64 Packages [985 B]
Get:126 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty/main i386 Packages [985 B]
Get:127 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty/main Translation-en [644 B]
Fetched 39.1 MB in 5s (6,605 kB/s)
Reading package lists...
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
The command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends $(travis_apt_get_options) install mysql-server-5.6 mysql-client-core-5.6 mysql-client-5.6 postfix" failed and exited with 100 during .
```

**Comment by ihor-sviziev** [comment]
I closed and re-opened PR just to trigger new build on Travis

**Review by aleron75 — APPROVED** [review]
Good work! Thanks for your contribution!

Once merged, don't forget you can easily port this PR to 2.2 branch with the Porting Tool, [read more here](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically).

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-3130](https://jira.corp.magento.com/browse/ENGCOM-3130) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@enl thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: lib/internal/Magento/Framework/Config/Test/Unit/_files/dom/big_cdata_text_new.xml
Comment (sidolov) [inline lib/internal/Magento/Framework/Config/Test/Unit/_files/dom/big_cdata_text_new.xml]: Please, replace comments with corresponded Magento copyrights

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by enl** [comment]
Hello @sidolov!

I fixed copyright comments in all created files but now build is failing on automated code review stage. And the error there is quite interesting: `The method _isCdataNode is not named in camelCase.`

Basically, it complains about method naming. I absolutely agree that the methods I've created should be private and camelCase but... There is one "BUT" here :) All other methods in this class have `protected` access level and start from "_" so that I consider this as naming convention.

What should I do? Break naming consistency for this class or ignore code review issues?

**Comment by aleron75** [comment]
> What should I do? Break naming consistency for this class or ignore code review issues?

not sure so I ask confirmation to @sidolov: maybe in this case we may add an annotation to ignore this kind of static analysis complaints?

**Comment by ihor-sviziev** [comment]
Hi @enl,
Right now we're using new code styles rules that requires not to use protected methods / properties in new code, the same for naming of methods / properties - they should not be prefixed with underscore.
Why we have quite a lot places that are containing such places - they were migraded from M1 and kept not touched for backward compatibility reasons. I know that it's confusing sometimes, but that's real situation that we have right now. At some point they should be deprecated and replaced, but that's quite a big list of such places, and we have to balance between fixing all of these issues and backward compatibility

**Comment by sidolov** [comment]
Hi @enl , changed files contain legacy code and old coding standards, you should try to avoid underscores in methods and properties names. All non-public methods should be private according to our rules that defined in [Technical Guidelines](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html#2-class-design) in section Class design. 
It's easy to fix this failures, just make methods private and remove underscore from names.
Thank you!

**Comment by enl** [comment]
Hi @sidolov any news on this?

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-3130](https://jira.corp.magento.com/browse/ENGCOM-3130) has been created to process this Pull Request

**Comment by sidolov** [comment]
Hi @enl , we tried to reproduce described issue on the 2.3-develop branch and looks like it works as expected. 

Steps to reproduce: 

1. Create custom module
2. Add system.xml to app/code/[Custom]/[Module]/etc/adminhtml/

```
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Config:etc/system_file.xsd">
    <system>
        <section id="web" translate="label" type="text" sortOrder="20" showInDefault="1" showInWebsite="1" showInStore="1">
            <label>Web</label>
            <tab>general</tab>
            <resource>Magento_Config::web</resource>
            <group id="url" translate="label" type="text" sortOrder="3" showInDefault="1" showInWebsite="1" showInStore="1">
                <label>Url Options</label>
                <field id="use_store" translate="label comment" type="select" sortOrder="10" showInDefault="1" showInWebsite="0" showInStore="0" canRestore="1">
                    <label>Add Store Code to Urls</label>
                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                    <backend_model>Magento\Config\Model\Config\Backend\Store</backend_model>
                    <comment>
                        <![CDATA[<strong style="color:red">Warning!</strong> When using Store Code in URLs, in some cases system may not work properly if URLs without Store Codes are specified in the third-party services (e.g. PayPal etc.)UPDATED.]]>
                    </comment>
                </field>
                <field id="redirect_to_base" translate="label comment" type="select" sortOrder="20" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
                    <label>Auto-redirect to Base URL</label>
                    <source_model>Magento\Config\Model\Config\Source\Web\Redirect</source_model>
                    <comment>I.e. redirect from http://example.com/store/ to http://www.example.com/store/</comment>
                </field>
            </group>
        </section>
    </system>
</config>
```

3. Navigate to Stores->Configuration->Web
4. Check comment message under Add Store Code to Urls field
5. Comment is updated


Maybe we missed something, please, add additional info to reproduce it. 
Thank you!

**Comment by enl** [comment]
Hello @sidolov! 

I've updated my fork to get all the changes from upstream and then performed the following steps:

1. `git clone https://github.com/enl/magento2.git --single-branch --branch 2.3-develop`
2. `composer install`
3. `bin/magento setup:install`
4. `composer require fastly/magento2` (Actually, it is unnecessary, but it was my original intention: to rewrite one of the comments in their module so that I suspected that it could be the issue with this module)
5. Created simple `Vaimo_Test` module with this file in `etc/adminhtml/system.xml:

```xml
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Config:etc/system_file.xsd">
<system>
    <section id="system">
        <group id="full_page_cache">
            <group id="fastly" translate="label" showInDefault="1" showInWebsite="0" showInStore="0" sortOrder="615">
                <group id="fastly_advanced_configuration" sortOrder="50" translate="label comment" showInDefault="1" showInWebsite="0" showInStore="0">
                    <field id="geoip_action" translate="label comment" type="select" sortOrder="70" showInDefault="1" showInWebsite="0" showInStore="0">
                        <label>GeoIP Action</label>
                        <comment>
                            <![CDATA["Dialog" shows a modal dialog to select target store.<br /> "Redirect" redirects the client directly to the target store. <br />Hello, world!]]>
                        </comment>
                        <source_model>Fastly\Cdn\Model\System\Config\Source\GeoIP\Action</source_model>
                        <depends>
                            <field id="enable_geoip">1</field>
                        </depends>
                    </field>
                </group>
            </group>
        </group>
    </section>
    <section id="web" translate="label" type="text" sortOrder="20" showInDefault="1" showInWebsite="1" showInStore="1">
        <label>Web</label>
        <tab>general</tab>
        <resource>Magento_Config::web</resource>
        <group id="url" translate="label" type="text" sortOrder="3" showInDefault="1" showInWebsite="1" showInStore="1">
            <label>Url Options</label>
            <field id="use_store" translate="label comment" type="select" sortOrder="10" showInDefault="1" showInWebsite="0" showInStore="0" canRestore="1">
                <label>Add Store Code to Urls</label>
                <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                <backend_model>Magento\Config\Model\Config\Backend\Store</backend_model>
                <comment>
                    <![CDATA[<strong style="color:red">Warning!</strong> When using Store Code in URLs, in some cases system may not work properly if URLs without Store Codes are specified in the third-party services (e.g. PayPal etc.)UPDATED.]]>
                </comment>
            </field>
            <field id="test" translate="label comment" type="text" sortOrder="15" showInDefault="1" showInStore="1" showInWebsite="1">
                <label>asdf</label>
                <comment>If you see me, it means that module's system.xml was loaded.</comment>
            </field>
            <field id="redirect_to_base" translate="label comment" type="select" sortOrder="20" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
                <label>Auto-redirect to Base URL</label>
                <source_model>Magento\Config\Model\Config\Source\Web\Redirect</source_model>
                <comment>I.e. redirect from http://example.com/store/ to http://www.example.com/store/</comment>
            </field>
        </group>
    </section>
</system>
</config>
```
6. `bin/magento setup:upgrade`

And I did no get field comments updated, but I can see my test "asdf" field:

Configuration___Settings___Stores___Magento_Admin


That makes it even more interesting, doesn't it? :)

My PHP version is:

```
PHP 7.2.10-0ubuntu0.18.04.1 (cli) (built: Sep 13 2018 13:45:02) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.10-0ubuntu0.18.04.1, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.0, Copyright (c) 2002-2018, by Derick Rethans
```

**Comment by sidolov** [comment]
Hi @enl , let's check it again. Thanks for the update

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed 

**Before:**

before

**After:**

after

**Pull request merged**
